tests/operators/test_constraints.py

```diff
@@ -84,6 +95,54 @@ def test_generate_level_constraint_multi_level():
     assert expected_pressure_constraint in repr(pressure_constraint)
 
 
+def test_generate_level_constraint_returnsinglelevel(load_verticalcoord_cubelist):
+    """For a cubelist that contains 3 cubes on different vertical levels.
+
+    Return one without a vertical coordinate.
+    """
+    constraint_1 = constraints.generate_level_constraint(
+        coordinate="pressure", levels=[]
+    )
+    constraint_2 = constraints.generate_level_constraint(
+        coordinate="model_level_number", levels=[]
+    )
+    combined = constraints.combine_constraints(constraint_1, a=constraint_2)
+
+    extracted = load_verticalcoord_cubelist.extract(combined)[0]
+
+    assert str(
+        extracted.coord
+    ), "<bound method Cube.coords of <iris 'Cube' of y_wind / (m s-1) (latitude: 2; longitude: 2)>>"
```

Comment:
You need to compare equality here. The expression after the comma is used when it raises its AssertationError. See https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement